### DISCUSSION TITLE
Duda tiempo y complejidad tests 6

### DISCUSSION THREAD

**Discussion by Segismundios** [IIC2133-PUC/2024-1#83] [Tarea 1] [answered]
Hola, me preguntaba si hay alguna forma recomendada para medir el tiempo de ejecución del programa. También me preguntaba si cuando se dice que medio segundo ya es mucho, es el programa entero? considerando la creación de los árboles y todo eso? o sólo se consideran las consultas. En caso de sólo considerarse las consultas, cómo se mediría ese tiempo?

**Comment by sachondo7** [comment, marked as answer]
con la librería time puedes medir el tiempo de ejecución. Acá hay un [ejemplo](https://barcelonageeks.com/como-medir-el-tiempo-que-tarda-una-funcion-en-c/). Eso sí, bórrala al entregar tu tarea final porque puede traer errores al corregir. 
Lo del medio segundo se tomó como un benchmark para cuando se itera 2 veces por las cabezas (1 para contar y otra para los resultados) lo

**Reply by Segismundios** [reply]
Creo que se cortó tu respuesta. Muchas gracias por lo del time, me funcionó bien. Igual me sigue quedando la duda, entonces corro el tiempo desde después de haber armado todos los árboles?

**Comment by sachondo7** [comment]
Lo del tiempo 0.5s es solo un número como para tener una idea nomás. Para optar por el bonus tienes que cumplir con **insertar y buscar en log(n)**. Lo importante no es el tiempo como tal, sino que cumplas la complejidad, se entiende?

**Reply by Segismundios** [reply]
Entonces la creación de los árboles (por la insersión) igual se evalúa que esté lo más optimizado posible? Había entendido que esa exigencia iba enfocada en la búsqueda y entrega de resultados. Lo del tiempo lo pregunto porque al menos en mi programa, la parte de creación de los árboles suma harto tiempo, entonces cambia bastante si mido la ejecución de todo el código vs a medir la ejecución desde que se realizan las búsquedas en adelante.

**Reply by sebacarrascop** [reply]
hola, esto queda más claro en la issue #87
